crates/optimism/node/src/engine.rs

```diff
@@ -101,11 +116,28 @@ impl PayloadValidator for OpEngineValidator {
         let sealed_block = self.inner.ensure_well_formed_payload(payload)?;
         sealed_block.try_recover().map_err(|e| NewPayloadError::Other(e.into()))
     }
+
+    fn validate_block_post_execution_with_hashed_state(
+        &self,
+        state_updates: &HashedPostState,
+        block: &RecoveredBlock<Self::Block>,
+    ) -> Result<(), ConsensusError> {
+        let state = self.provider.latest().map_err(|err| {
```

Comment:
I believe this is problematic, because latest isn't necessarily correct, so we need the state for `block.parent`